src/github.com/getlantern/flashlight/genconfig/checkfallbacks.go

```diff
@@ -0,0 +1,166 @@
+// Utility for checking if fallback servers are working properly.
+// It outputs failing servers info in STDOUT.  This allows this program to be
+// used for automated testing of the fallback servers as a cron job.
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"runtime"
+	"strings"
+	"sync"
+
+	"github.com/getlantern/golog"
+
+	"github.com/getlantern/flashlight/client"
+)
+
+var (
+	help          = flag.Bool("help", false, "Get usage help")
+	fallbacksFile = flag.String("fallbacks", "fallbacks.json", "File containing json array of fallback information")
+	verbose       = flag.Bool("verbose", false, "Verbose output for debugging")
+	numConns      = flag.Int("connections", 1, "Number of simultaneous connections")
+
+	expectedBody = "Google is built by a large team of engineers, designers, researchers, robots, and others in many different sites across the globe. It is updated continuously, and built with more tools and technologies than we can shake a stick at. If you'd like to help us out, see google.com/careers.\n"
+)
+
+var (
+	log = golog.LoggerFor("checkfallbacks")
+)
+
+type fallbackServer map[string]interface{}
+
+func main() {
+	flag.Parse()
+
+	if *help {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	numcores := runtime.NumCPU()
+
+	if *verbose {
+		log.Debugf("Using all %d cores on machine", numcores)
+	}
+	runtime.GOMAXPROCS(numcores)
+
+	fallbacks := make(chan fallbackServer)
+	loadFallbacks(fallbacks)
+	testFallbacks(fallbacks)
+}
+
+// Load the fallback servers list file. Failure to do so will result in
+// exiting the program.
+func loadFallbacks(fallbacks chan<- fallbackServer) {
+	if *fallbacksFile == "" {
+		log.Error("Please specify a fallbacks file")
+		flag.Usage()
+		os.Exit(2)
+	}
+
+	fallbacksBytes, err := ioutil.ReadFile(*fallbacksFile)
+	if err != nil {
+		log.Fatalf("Unable to read fallbacks file at %s: %s", *fallbacksFile, err)
+		os.Exit(2)
+	}
+
+	var unmarshalledFallbacks []fallbackServer
+	err = json.Unmarshal(fallbacksBytes, &unmarshalledFallbacks)
+	if err != nil {
+		log.Fatalf("Unable to unmarshal json from %v: %v", *fallbacksFile, err)
+		os.Exit(1)
+	}
+
+	// Channel fallback servers on-demand
+	go func() {
+		for _, val := range unmarshalledFallbacks {
+			fallbacks <- val
+		}
+		close(fallbacks)
+	}()
+}
+
+// Test fallback servers. Output to stdout *only* when a test fails.
+func testFallbacks(fallbacks <-chan fallbackServer) {
+	workersWg := sync.WaitGroup{}
+
+	if *verbose {
+		fmt.Printf("Spawning %d workers\n", *numConns)
+	}
+
+	workersWg.Add(*numConns)
+	for i := 0; i < *numConns; i++ {
+		// Worker: consume fallback servers from channel and signal
+		// Done() when closed (i.e. range exits)
+		go func(i int) {
+			for fb := range fallbacks {
+				testFallbackServer(fb, i)
+			}
+			workersWg.Done()
+		}(i+1)
+	}
+
+	workersWg.Wait()
```

Comment:
And here you could close the output channel.  You could run testFallbacks in a goroutine and have the main goroutine consume the output channel, so when it gets closed the program can exit.